### DISCUSSION TITLE
ESP32-S3 supported?

### DISCUSSION THREAD

**Discussion by samuk** [makerspet/support#17] [General]
Hey this looks good.

I'd like to try it on https://rosmo-robot.github.io/

I think it looks broadly compatible? Particularly if I just use two motors for 2wd?

**Comment by kaiaai** [comment]
Hello,
  sure, give it a try. It supports ESP32-S3 as well as N20 encoder motors.
Try using the "kaiaai/firmware config" branch. I haven't released it yet (it needs documentation), but it's designed to work with 3rd party/custom boards like yours. There is a config.yaml file where you put your board GPIO numbers (motor in1/in2, encoder A/B, Lidar tx/rx/etc). This makes adding new boards a breeze, no need to touch the firmware.
  That said, your robot mentions IMU and camera. There is no support for IMU or camera for now. This is because microROS bandwidth over WiFi UDP is quite limited. SLAM/mapping works fine without IMU.

**Reply by samuk** [reply]
Nice, I'll give it a go. The camera isn't integrated (yet) it's just a Xiao camera powered by the robot battery streaming over its own connection to the network.

In the current Ros2 implementation: https://github.com/rosmo-robot/linorobot2_hardware/tree/master/firmware (Linorobot2 fork) the IMU processing happens on the ESP32.

**Reply by samuk** [reply]
I was interested to see you're using dockerised ROS I started editing https://github.com/Tiryoh/docker-ros-desktop-vnc so users got a browser based webui to Ubuntu.

Perhaps we could collaborate on that? Maintain a single image with makerspet/ linorobot workspaces?

**Reply by kaiaai** [reply]
Yes, I do use dockerised ROS https://github.com/kaiaai/install
I run the Docker image locally on my Windows laptop and all Ubuntu GUI shows up nicely using an X server. So, there is no need for VNC or VNC-in-a-browser as long as the PC is local.
Thanks for the firmware link. I can see IMU, mag, battery, range, odom getting published over micro-ROS and Lidar published over a UDP server.
About a single Docker image - this is a company project. You're welcome to use all our open source, it is liberally licensed (Apache 2.0), but bringing in a 3rd party project into this particular Docker image is typically (with rare exceptions) a no-no.  That said, it should be quite straightforward to build a Docker image on top of our docker image.
